### PR TITLE
Implements the RegressionChannel indicator

### PR DESCRIPTION
**Pull request by AlexCatarino**
The Regression Channel indicator extends the LeastSquaresMovingAverage with the inclusion of two (upper and lower) channel lines that are distanced from the linear regression line by a user defined number of standard deviations.
Ref.: #474

### CODE REVIEW

File: Indicators/LeastSquaresMovingAverage.cs [hunk 1]
```diff
@@ -63,23 +75,31 @@ public LeastSquaresMovingAverage(int period)
...
+                .OrderBy(i => i.Time)
+                .Select(i => Convert.ToDouble(i.Value))
+                .ToArray();
+            // Fit OLS
+            var ols = Fit.Line(x: t, y: series);
+            Intercept.Update(input.Time, (decimal)ols.Item1);
+            Slope.Update(input.Time, (decimal)ols.Item2);
+
+            // Calculate the fitted value corresponding to the input
+            return Intercept + Slope * Period;
```
Comment (mchandschuh) [inline Indicators/LeastSquaresMovingAverage.cs]: Looks like you already fixed it :) 

You could squash the commit that fixed this into the commit that produced it. Not required, but nice easy way of dipping your toes into the `git rebase` world of goodness :)

Comment (AlexCatarino) [inline Indicators/LeastSquaresMovingAverage.cs]: In my first try, Intercept and Slope were not window indicator, but then I used Intercept.Plus(Slope.Times(Period)), it was returning zeroes. Then I moved to window and use that expression. Finally, when you suggested reviewing the usage of window indicator, I re-access everything and come up with this solution.

I will wait for final reviews and squash some commits (if everyone agrees) that were mostly typos and variable renaming.

Comment (mchandschuh) [inline Indicators/LeastSquaresMovingAverage.cs]: Variable renames are nice in their own commits. Ideally,  any change that has no effect on behavior should be in its own commit. This includes doc changes,  white space changes, renames and mechanical refactorings. This helps the diff engine during merges and helps the poor sap that needs to look through the history to find out how something for some way... It's much better to be able to completely ignore the renames in the diff and instead look at a change that is primarily changing behavior, vs a commit that changes behaviors and has renames and refactorings, ect...

Comment (AlexCatarino) [inline Indicators/LeastSquaresMovingAverage.cs]: Thanks for the pro-tip.

**Comment by jaredbroad** [comment]
Thanks Alex

**Pull request merged**